staging/src/k8s.io/client-go/tools/auth/exec/exec_test.go

```diff
@@ -57,7 +57,8 @@ func TestLoadExecCredential(t *testing.T) {
 						CertificateAuthorityData: []byte("some-ca-data"),
 						ProxyURL:                 "https://some-proxy-url:12345",
 						Config: runtime.RawExtension{
-							Raw: []byte(`{"apiVersion":"group/v1","kind":"PluginConfig","spec":{"names":["marshmallow","zelda"]}}`),
+							Raw:         []byte(`{"apiVersion":"group/v1","kind":"PluginConfig","spec":{"names":["marshmallow","zelda"]}}`),
+							ContentType: runtime.ContentTypeJSON,
```

Comment:
/assign enj
per triage:
@enj "Do we actually expect the content to be JSON or just a blob of data we don't parse?"

Reply:
I don't recall if we ever enforced any kind of encoding requirements on the config we pass to exec plugins, so it is unclear to me what kind of inputs we allowed before.  Whatever we did allow before needs to keep working as-is.

Reply:
The proposal is https://github.com/kubernetes/enhancements/pull/4707#discussion_r1638814864. If a RawExtension is populated via calling UnmarshalJSON, we know for certain at that time that the raw bytes are JSON-encoded. The content type field would never be serialized and serves only as a hint to avoid heuristics in cases where the content type is known definitively. In this case it's a programmatic assignment of some JSON bytes to Raw, so since we know the encoding it's reasonable to also set a content type. It doesn't change expectations about the encoding of this RawExtension field.

This is on hold to consider roundtrippability issues related to not serializing the content type (essentially, RawExtension -> JSON -> RawExtension would result in a populated ContentType and RawExtension -> Proto -> RawExtension would not).

Reply:
I think you are saying that exec scenario (linked below) would not be impacted?

https://github.com/kubernetes/kubernetes/blob/970dc8c2c2af485bc8d9b4eec4669e97de7e2359/staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1/types.go#L104-L126